src/hashtree.erl

```diff
@@ -413,14 +457,21 @@ width(#state{width=W}) ->
 mem_levels(#state{mem_levels=M}) ->
     M.
 
+-spec path(hashtree()) -> string().
+path(#state{path=P}) ->
+    P.
+
 %% Note: meta is currently a one per file thing, even if there are multiple
 %%       trees per file. This is intentional. If we want per tree metadata
 %%       this will need to be added as a separate thing.
--spec write_meta(binary(), binary(), hashtree()) -> hashtree().
+-spec write_meta(binary(), binary()|term(), hashtree()) -> hashtree().
 write_meta(Key, Value, State) when is_binary(Key) and is_binary(Value) ->
     HKey = encode_meta(Key),
     ok = eleveldb:put(State#state.ref, HKey, Value, []),
-    State.
+    State;
+write_meta(Key, Value0, State) when is_binary(Key) ->
```

Comment:
nvm previous comment, i get your point @jonmeredith. Do we want to roll some sort of ets+dets localized reader/writer thing (storing per tree-id) and not tying the counter to level? Or is that fraught w/ much peril? :).